haystack/preview/components/preprocessors/document_language_classifier.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import List, Dict, Any, Optional
+
+from haystack.preview import component, default_from_dict, default_to_dict, Document
+from haystack.preview.lazy_imports import LazyImport
+
+logger = logging.getLogger(__name__)
+
+with LazyImport("Run 'pip install langdetect'") as langdetect_import:
+    import langdetect
+
+
+@component
+class DocumentLanguageClassifier:
+    """
+    Routes documents onto different output connections depending on their language.
+    This is useful for routing documents to different models in a pipeline depending on their language.
+    The set of supported languages can be specified.
+    For routing texts based on their language use the related TextLanguageClassifier component.
+
+    Example usage in and indexing pipeline that writes only English language documents to a Store:
+
+    ```python
+    document_store = MemoryDocumentStore()
+    p = Pipeline()
+    p.add_component(instance=TextFileToDocument(), name="text_file_converter")
```

Comment:
Not strictly related to this PR, but why inverting the order of the original signature:
```
def add_component(self, name: str, instance: Component) -> None:
```
if passing the instance first is more intuitive we're on time to change it!

Reply:
I think it's more intuitive, yes. I created an issue here: https://github.com/deepset-ai/canals/issues/137